home-assistant-plugin/custom_components/xknx/schema.py

```diff
@@ -139,13 +139,31 @@ class LightSchema:
     DEFAULT_MIN_KELVIN = 2700  # 370 mireds
     DEFAULT_MAX_KELVIN = 6000  # 166 mireds
 
+    RED = "red"
+    GREEN = "green"
+    BLUE = "blue"
+    WHITE = "white"
+
+    COLOR_SCHEMA = vol.Schema(
+        {
+            vol.Optional(CONF_ADDRESS): cv.string,
+            vol.Optional(CONF_STATE_ADDRESS): cv.string,
+            vol.Optional(CONF_BRIGHTNESS_ADDRESS): cv.string,
+            vol.Optional(CONF_BRIGHTNESS_STATE_ADDRESS): cv.string,
+        }
+    )
+
     SCHEMA = vol.Schema(
         {
             vol.Required(CONF_ADDRESS): cv.string,
             vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
             vol.Optional(CONF_STATE_ADDRESS): cv.string,
             vol.Optional(CONF_BRIGHTNESS_ADDRESS): cv.string,
             vol.Optional(CONF_BRIGHTNESS_STATE_ADDRESS): cv.string,
+            vol.Optional(RED): COLOR_SCHEMA,
+            vol.Optional(GREEN): COLOR_SCHEMA,
+            vol.Optional(BLUE): COLOR_SCHEMA,
+            vol.Optional(WHITE): COLOR_SCHEMA,
             vol.Optional(CONF_COLOR_ADDRESS): cv.string,
             vol.Optional(CONF_COLOR_STATE_ADDRESS): cv.string,
```

Comment:
I think red green blue be in an inclusion group (optional white) and 
these 3/4 in an exclusion group with color_address. https://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html#module-voluptuous.schema_builder

Reply:
https://github.com/XKNX/xknx/pull/508/commits/15cf578a43a8d07dcfa6f24a3f70ce410f1aaa55
I have tried to put the color addresses in an exclude group, of course without changing existing behaviour. I couldn't add the exclude to both the color and color_state addresses because it would make it impossible to use them both at the same time, as should be possible.